python/HeaderService/camera_coords.py

```diff
@@ -253,62 +287,170 @@ def mosaic_ITL(self, Sx, Sy):
     def get_channel(self, Segment):
         return CHANNEL[self.vendor][Segment]
 
-    def wcs_tan_matrix(self, ra, dec, rot_angle=0):
+    def wcs_TAN(self, ra, dec, rot_angle, radesys):
 
         '''
         Compute the common section of the WCS Tangential projection matrix
         See for a definition http://danmoser.github.io/notes/gai_fits-imgs.html
         '''
 
-        CDELT1 = self.xpixelscale
-        CDELT2 = self.ypixelscale,
-        CROTA2 = rot_angle
         wcs = {'CTYPE1': 'RA---TAN',
                'CTYPE2': 'DEC--TAN',
                'CUNIT1': 'deg',
                'CUNIT2': 'deg',
                'CRVAL1': ra,
                'CRVAL2': dec,
-               # ------------------------------------------
-               # We will use a function for these later on
-               # 'CRPIX1' : 5,
-               # 'CRPIX2' : 10,
-               # ------------------------------------------
-               'CROTA2': rot_angle,
-               'CDELT1': self.xpixelscale,
-               'CDELT2': self.ypixelscale,
-               'CD1_1': CDELT1*math.cos(CROTA2),
-               'CD1_2': -CDELT2*math.sin(CROTA2),
-               'CD2_1': CDELT1*math.sin(CROTA2),
-               'CD2_2': CDELT2*math.cos(CROTA2),
+               'CRPIX1': self.CRPIX1,
+               'CRPIX2': self.CRPIX2,
+               'CD1_1': self.CDELT1*math.cos(rot_angle),
+               'CD1_2': -self.CDELT2*math.sin(rot_angle),
+               'CD2_1': self.CDELT1*math.sin(rot_angle),
+               'CD2_2': self.CDELT2*math.cos(rot_angle),
+               'RADESYS': radesys,
                }
         return wcs
 
-    def wcs_CCD(self, Sx, Sy):
+    def wcs_C(self, Sx, Sy):
+
+        """
+        This function defines the WCS representation for a Segment (Sx,Sy) in
+        CCD coordinates as described in section 9.6 of LCA-13501 for e2V and
+        ITL sensors.
+        """
+
+        if self.vendor == 'E2V':
+            wcs = {'PC1_1C': 0,
+                   'PC1_2C': 1 - 2*Sx,
+                   'PC2_1C': 1 - 2*Sx,
+                   'PC2_2C': 0,
+                   'CDELT1C': 1,
+                   'CDELT2C': 1,
+                   'CRPIX1C': 0,
+                   'CRPIX2C': 0,
+                   'CRVAL1C': Sx*(2*self.dimv + 1),
+                   'CRVAL2C': Sx*(self.dimh + 1) + Sy*self.dimh+(2*Sx - 1)*self.preh}
+        elif self.vendor == 'ITL':
+            wcs = {'PC1_1C': 0,
+                   'PC1_2C': 1 - 2*Sx,
+                   'PC2_1C': -1,
+                   'PC2_2C': 0,
+                   'CDELT1C': 1,
+                   'CDELT2C': 1,
+                   'CRPIX1C': 0,
+                   'CRPIX2C': 0,
+                   'CRVAL1C': Sx*(2*self.dimv+1),
+                   'CRVAL2C': self.dimh + 1 + Sy*self.dimh - self.preh}
+        return wcs
+
```

Comment:
Are you reading the vendor from CCS (see above)?
Why are we writing a "C" WCS?
Why are we using `PCn_m` keys here and `CDn_m` earlier; we should use one or the other consistently.  The standard forbids us from mixing `CD` and `PC` cards in a single HDU, and while it's probably OK this formally forbids us to mix-and-match even with `WCS` and `WCSC`

Reply:
Although these are not used, I've keep the functions in case Tony wants this back in the future. If you want me to removed them, say so, and I will.